From 122a4bed3c50a217e6fdd8e929c29be97ca8e59e Mon Sep 17 00:00:00 2001
From: ricolin <rlin@vexxhost.com>
Date: Tue, 9 Sep 2025 00:29:21 +0800
Subject: [PATCH] [Neutron] ovn adopt uwsgi

as [1] fixed and [2] introduced, we should use uwsgi for Neutron server
with OVN backend now.

Also fix config missing.

[1] https://bugs.launchpad.net/neutron/+bug/1912359
[2] https://bugs.launchpad.net/neutron/+bug/2087953

Change-Id: Ic86a4d3a58fb916ef40c33261e4472e0d3e5457b
Signed-off-by: ricolin <rlin@vexxhost.com>
---
 .../templates/bin/_neutron-server.sh.tpl      | 28 ++++++-------------
 1 file changed, 9 insertions(+), 19 deletions(-)

diff --git a/neutron/templates/bin/_neutron-server.sh.tpl b/neutron/templates/bin/_neutron-server.sh.tpl
index be4b254a..157a3dc4 100644
--- a/neutron/templates/bin/_neutron-server.sh.tpl
+++ b/neutron/templates/bin/_neutron-server.sh.tpl
@@ -18,36 +18,26 @@ set -ex
 COMMAND="${@:-start}"
 
 function start () {
-# (ricolin): Currently ovn have issue with uWSGI,
-# let's keep using non-uWSGI way until this bug fixed:
-# https://bugs.launchpad.net/neutron/+bug/1912359
+  confs="--config-file /etc/neutron/neutron.conf"
 {{- if ( has "ovn" .Values.network.backend ) }}
-  start_ovn
-{{- else }}
-  exec uwsgi --ini /etc/neutron/neutron-api-uwsgi.ini
-{{- end }}
-}
-
-function start_ovn () {
-  exec neutron-server \
-        --config-file /etc/neutron/neutron.conf \
-{{- if ( has "ovn" .Values.network.backend ) }}
-        --config-file /tmp/pod-shared/ovn.ini \
+  confs+=" --config-file /tmp/pod-shared/ovn.ini"
 {{- end }}
 {{- if .Values.conf.plugins.taas.taas.enabled }}
-        --config-file /etc/neutron/taas_plugin.ini \
+  confs+=" --config-file /etc/neutron/taas_plugin.ini"
 {{- end }}
 {{- if ( has "sriov" .Values.network.backend ) }}
-        --config-file /etc/neutron/plugins/ml2/sriov_agent.ini \
+  confs+=" --config-file /etc/neutron/plugins/ml2/sriov_agent.ini"
 {{- end }}
 {{- if .Values.conf.plugins.l2gateway }}
-        --config-file /etc/neutron/l2gw_plugin.ini \
+  confs+=" --config-file /etc/neutron/l2gw_plugin.ini"
 {{- end }}
 {{- if ( has "tungstenfabric" .Values.network.backend ) }}
-        --config-file /etc/neutron/plugins/tungstenfabric/tf_plugin.ini
+  confs+=" --config-file /etc/neutron/plugins/tungstenfabric/tf_plugin.ini"
 {{- else }}
-        --config-file /etc/neutron/plugins/ml2/ml2_conf.ini
+  confs+=" --config-file /etc/neutron/plugins/ml2/ml2_conf.ini"
 {{- end }}
+
+  exec uwsgi --ini /etc/neutron/neutron-api-uwsgi.ini --pyargv " $confs "
 }
 
 function stop () {
-- 
2.25.1

